### PR TITLE
fetch products + collections by handle

### PR DESCRIPTION
**Pull request by lyninx**
exposing the storefront api's ability to fetch products and collections by handle. `Shop` has `productByHandle` and `collectionByHandle`.

this feature is required to support existing buy-buttons, which are configured to load products/collections using handles.

related bb.js issue: https://github.com/Shopify/buy-button-js/issues/400

### CODE REVIEW

**Comment by swalkinshaw** [comment]
I don't think the `shop` prefix is needed. I mean technically everything is scoped to shop.

File: src/collection-by-handle-query.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import addFields from './add-fields';
+import productConnectionQuery from './product-connection-query';
+import imageQuery from './image-query';
+
+const defaultFields = ['id', 'handle', 'description', 'descriptionHtml', 'updatedAt', 'title', ['products', productConnectionQuery()], ['image', imageQuery()]];
```
Comment (minasmart) [inline src/collection-by-handle-query.js]: line breaks please.

File: src/client.js [hunk 2]
```diff
@@ -243,6 +247,60 @@ class Client {
   }
 
   /**
+   * Fetches a single product by handle on the shop.
+   *
+   * @example
+   * client.fetchProductByHandle('my-produt').then((product) => {
```
Comment (swalkinshaw) [inline src/client.js]: typo here: `my-product`

File: src/client.js [hunk 3]
```diff
@@ -243,6 +247,60 @@ class Client {
...
+        product.attrs.variants = variants;
+      }));
+
+      return Promise.all(promises).then(() => {
+        return product;
+      });
+    });
+  }
+
+  fetchCollectionByHandle(handle) {
```
Comment (swalkinshaw) [inline src/client.js]: Should this function have docs too?

Comment (minasmart) [inline src/client.js]: It should.

**Comment by minasmart** [comment]
also, maybe interactive rebase and flatten this PR out a bit.

File: src/client.js (hunk 3 above)
Comment (lyninx) [inline src/client.js]: oops. yeah totally forgot about that!

**Pull request merged**